### ISSUE TITLE
S4457 False positive when argument check is after async code

### ISSUE THREAD

**Issue by paulhickman-a365** [SonarSource/sonar-dotnet#6449]
### Description

In Sonar Analyzer version 8.49, the following code, the method HasS4457Async erroneously raises S4457, whereas this does not happen for the method NoS4457Async

The method cannot be split to do the validation before the async code as the validation relies on the result of the async code.

In Sonar Analyzer version 8.48, it doesn't cause S4457.

### Repro steps

````c#
    public class Class1
    {
        public Task<int[]> CheckAsync() => Task.FromResult(new int[] { 1 });
        public Task<int> Check2Async() => Task.FromResult(1);

        public async Task HasS4457Async(int request)
        {
            var identifierType = (await CheckAsync()).FirstOrDefault(x => x == request);
            if (identifierType == 0) throw new ArgumentException("message");
        }

        public async Task NoS4457Async(int request)
        {
            var identifierType = await Check2Async();
            if (identifierType != request) throw new ArgumentException("message");
        }
    }
````
### Expected behavior

No warning

### Actual behavior

Warning S4457

### Known workarounds

Downgrade to 8.48 or suppress the warning

### Related information

* Project Type NET Standard 2.1 class library
* Visual Studio 16.4.1
* SonarScanner for .NET version 8.49.8.49.0.57237
* Windows Server 2022

**Comment by mary-georgiou-sonarsource** [comment]
Thank you for opening this issue, @paulhickman-a365. Your feedback helps us improve our products. We will look at it as soon as possible. After we confirm the issue, we will add labels and put it in our backlog.

**Comment by gregory-paidis-sonarsource** [comment]
Hello @paulhickman-a365, thanks for the feedback, very nice catch.

The necessary work has been completed and this FP will be resolved after the next release.
